src/domain/CreateEventStreamRecordRepository.ts

```diff
@@ -11,3 +14,20 @@ export type CreateEventStreamRecord = AuditRecord & {
 };
 
 export type CreateEventStreamRecordRepository = Repository<CreateEventStreamRecord>;
+
+export const isCreateEventStreamRecord = (record: AllRecord): O.Option<CreateEventStreamRecord> =>
+  record.type === 'CreateEventStreamRecord' ? O.some(record) : O.none;
+
+export const hasTimelineEventTypeToTimeline = (record: CreateEventStreamRecord) =>
+  record.input.body.eventType === EventTypeEnum.TIMELINE;
+
+export const isSuccessfulResponse = (record: CreateEventStreamRecord) => record.output.statusCode === 200;
+
+export const existsCreateEventStreamRecordWhitStreamId =
```

Comment:
At some point, we should find a consistent place for these predicates. They are sometimes in the `*Repository` files, and sometimes in the `checks/*Checks` files :persevere:

Reply:
I agree! We could track this in a future work